source/docs/magento2/detect-admin-users.md

```diff
@@ -0,0 +1,178 @@
+import { AdminDataLoader } from "./loaders";
+
+-export const AdminDataLoader = (axiosInstance) => {
++export const AdminDataLoader = (axiosInstance, MagentoAdmin) => {
+  return {
+    loadData: async () => {
++     if (await MagentoAdmin.isAdmin()) {
+-       const response = await axiosInstance.get("/admin-custom-endpoint")
++       const response = await axiosInstance.get("/admin-custom-endpoint", {
++         headers: {
++           // We pass a header here considering that the `/admin-custom-endpoint`
++           // now only accepts requests with this header
++           "x-custom-header": "your custom header matching a custom configuration in your Magento"
++         }    
++       })
+        return response.data
++     } else {
++       return null;
++     }
+    },
+  };
+};
+```
+
+<blockquote class="important">
+**Important:** In case you rely on [Front-Commerce's caching mechanism](/docs/advanced/graphql/dataloaders-and-cache-invalidation.html) you will need to make sure that data fetched by admins are not cached in the same namespace as your normal users.
+For this reason, please make sure to update your [Caching strategies](/docs/advanced/graphql/dataloaders-and-cache-invalidation.html#Caching-strategies) with the relevant configurations.
+</blockquote>
+
+## Change the behavior depending on the admin connected
+
+In some case you may need more granularity in your authentication process. In the above section, all admins are considered equals. But in practice, there might be some roles or custom authentication process.
+
+This can be done in Front-Commerce by customizing the data available in `loaders.MagentoAdmin.getAdminData()`. By default it is an empty object when the admin is connected. But this object can be configured by injecting a custom `AuthenticatedAdminDataInterface` in your Magento DI.
+
+To do so, in a custom Magento module, you must:
+
+* Add a new preference in your `etc/di.xml`:
+```xml
+<preference for="FrontCommerce\AdminBar\Api\AuthenticatedAdminDataInterface" type="Acme\ModuleName\Model\AuthenticatedAdminData" />
+```
+* Create the new class that will implement `AuthenticatedAdminDataInterface` and fetch the addition admin data. Since you are in a Magento context, feel free to inject any class you may need to fetch the relevant data. To keep it simple here, we've just set a hardcoded id.
+```php
+<?php
+namespace Acme\ModuleName\Model;
+
+use FrontCommerce\AdminBar\Api\AuthenticatedAdminDataInterface;
+
+class AuthenticatedAdminData implements AuthenticatedAdminDataInterface
+{
+    public function getAdminData()
+    {
+        return [
+            'adminId' => '13'
+        ];
+    }
+}
+```
+* If you are not in `developer` mode in magento, please make sure to rerun `bin/magento setup:di:compile` command.
+* You will then be able to use the following in Front-Commerce's server code for any authenticated admin users.
+```js
+console.log(loaders.MagentoAdmin.getAdminData().adminId)
```

Comment:
```suggestion
const adminData = await loaders.MagentoAdmin.getAdminData();
console.log(adminData.adminId)
```

(related to https://gitlab.com/front-commerce/front-commerce/-/merge_requests/222#note_376339984)